--- origsrc/gst-python-0.10.21/gst/Makefile.am	2011-01-05 13:24:37.000000000 -0600
+++ src/gst-python-0.10.21/gst/Makefile.am	2011-07-07 02:38:54.514450800 -0500
@@ -104,7 +104,7 @@ pbutils_la_CFLAGS = $(common_cflags) $(G
 pbutils_la_LIBADD = $(common_libadd) $(GST_PLUGINS_BASE_LIBS) -lgstpbutils-0.10
 pbutils_la_LDFLAGS = $(common_ldflags) \
 	-export-symbols-regex "^(initpbutils|_PyGObject_API).*" $(GST_PLUGINS_BASE_LIBS)
-pbutils_la_SOURCES = pbutilsmodule.c gst-argtypes.c
+pbutils_la_SOURCES = pbutilsmodule.c gst-argtypes.c pygstminiobject.c
 nodist_pbutils_la_SOURCES = pbutils.c
 PBUTILS_OVERRIDES = pbutils.override
 PBUTILS_DEFS = pbutils.defs
--- origsrc/gst-python-0.10.21/testsuite/Makefile.am	2011-02-01 13:33:07.072434200 -0600
+++ src/gst-python-0.10.21/testsuite/Makefile.am	2011-02-01 07:29:32.000000000 -0600
@@ -8,7 +8,7 @@ INCLUDES = \
 linked_LIBS = testhelper.la
 
 testhelper_la_LDFLAGS = -module -avoid-version
-testhelper_la_LIBADD = $(GLIB_LIBS)
+testhelper_la_LIBADD = $(GST_LIBS)
 testhelper_la_CFLAGS = $(PYGOBJECT_CFLAGS)
 testhelper_la_SOURCES = \
 	testhelpermodule.c \
