--- origsrc/gst-python/gst/__init__.py	2007-06-14 09:53:28.000000000 -0500
+++ src/gst-python/gst/__init__.py	2008-04-25 08:11:32.921875000 -0500
@@ -119,40 +119,15 @@
         return float(self.num) / float(self.denom)
 
 
-import sys
-dlsave = sys.getdlopenflags()
+from _gst import *
+import interfaces
 try:
-    from DLFCN import RTLD_GLOBAL, RTLD_LAZY
-except ImportError:
-    RTLD_GLOBAL = -1
-    RTLD_LAZY = -1
-    import os
-    osname = os.uname()[0]
-    if osname == 'Linux' or osname == 'SunOS' or osname == 'FreeBSD':
-        RTLD_GLOBAL = 0x100
-        RTLD_LAZY = 0x1
-    elif osname == 'Darwin':
-        RTLD_GLOBAL = 0x8
-        RTLD_LAZY = 0x1
-    del os
+    import libxml2
 except:
-    RTLD_GLOBAL = -1
-    RTLD_LAZY = -1
-
-if RTLD_GLOBAL != -1 and RTLD_LAZY != -1:
-    sys.setdlopenflags(RTLD_LAZY | RTLD_GLOBAL)
-    from _gst import *
-    import interfaces
-    try:
-        import libxml2
-    except:
-        pass
+    pass
 
 version = get_gst_version
 
-sys.setdlopenflags(dlsave)
-del sys
-
 # Fixes for API cleanups that would cause an API breakage.
 # See #446674
 
--- origsrc/gst-python/gst/gst.override	2008-02-29 06:37:43.000000000 -0600
+++ src/gst-python/gst/gst.override	2008-04-25 08:11:32.921875000 -0500
@@ -584,6 +584,24 @@
     return xml;
  }
 
+#ifdef G_WITH_CYGWIN
+PyObject *
+libxml_xmlDocPtrWrap(xmlDocPtr doc)
+{
+    PyObject *ret;
+
+    if (doc == NULL) {
+        Py_INCREF(Py_None);
+        return (Py_None);
+    }
+    /* TODO: look at deallocation */
+    ret =
+        PyCObject_FromVoidPtrAndDesc((void *) doc, (char *) "xmlDocPtr",
+                                     NULL);
+    return (ret);
+}
+#endif
+
 static int
 _wrap_gst_xml_new(PyGObject *self)
 {
--- origsrc/gst-python/testsuite/Makefile.am	2008-01-01 07:22:21.000000000 -0600
+++ src/gst-python/testsuite/Makefile.am	2008-04-25 08:11:32.937500000 -0500
@@ -8,7 +8,7 @@
 linked_LIBS = testhelper.la
 
 testhelper_la_LDFLAGS = -module -avoid-version
-testhelper_la_LIBADD = $(GLIB_LIBS)
+testhelper_la_LIBADD = $(GST_LIBS)
 testhelper_la_CFLAGS = $(PYGOBJECT_CFLAGS)
 testhelper_la_SOURCES = \
 	testhelpermodule.c \
